File: app/src/main/java/cn/chenyuanming/gankmeizhi/utils/DbHelper.java
package cn.chenyuanming.gankmeizhi.utils;

import com.litesuits.orm.LiteOrm;

import java.util.ArrayList;
import java.util.List;

import cn.chenyuanming.gankmeizhi.application.GankApplication;

/**
 * Created by Administrator on 2016/1/28.
 */
public class DbHelper {
    private static DbHelper helper;

    private DbHelper() {
        liteOrm = LiteOrm.newSingleInstance(GankApplication.getContext(), "gankmeizhi.db");
    }

    private LiteOrm liteOrm;

    public static DbHelper getHelper() {
        if (helper == null) {
            helper = new DbHelper();
        }
        return helper;
    }

    public LiteOrm getLiteOrm() {
        return liteOrm;
    }


    public <T> List<T> getData(Class<T> clazz) {
        List<T> list = getHelper().getLiteOrm().query(clazz);
        if (list == null || list.size() == 0) {
            list = new ArrayList<>();
            try {
                list.add(clazz.newInstance());
                getHelper().getLiteOrm().save(list);
            } catch (InstantiationException e) {
                e.printStackTrace();
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            }
        }
        return list;
    }
}

File: app/src/main/java/cn/chenyuanming/gankmeizhi/utils/DbHelper.kt
package cn.chenyuanming.gankmeizhi.utils

import cn.chenyuanming.gankmeizhi.application.GankApplication
import com.litesuits.orm.LiteOrm
import java.util.*

/**
 * Created by Chen Yuanming on 2016/1/28.
 */
class DbHelper private constructor() {

    val liteOrm: LiteOrm

    init {
        liteOrm = LiteOrm.newSingleInstance(GankApplication.context, "gankmeizhi.db")
    }


    fun <T> getData(clazz: Class<T>): MutableList<T> {
        var list: MutableList<T>? = getHelper().liteOrm.query(clazz)
        if (list == null || list.size == 0) {
            list = ArrayList()
            try {
                list.add(clazz.newInstance())
                getHelper().liteOrm.save(list)
            } catch (e: InstantiationException) {
                e.printStackTrace()
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
            }

        }
        return list
    }

    companion object {
        private  var helper: DbHelper = DbHelper()

        fun getHelper(): DbHelper {
            return helper
        }
    }
}
